src/ansys/aedt/core/modeler/cad/primitives_3d.py

```diff
@@ -1464,6 +1464,56 @@ def _create_reference_cs_from_3dcomp(self, assignment, password):
             app.oproject.Close()
             return assignment.target_coordinate_system
 
+    @staticmethod
+    def __create_temp_project(app):
+        """Create temporary project with a duplicated design."""
+        temp_proj_name = generate_unique_project_name()
+        ipkapp_temp = Icepak(project=os.path.join(app.toolkit_directory, temp_proj_name))
+        ipkapp_temp.delete_design(ipkapp_temp.design_name)
+        app.oproject.CopyDesign(app.design_name)
+        ipkapp_temp.oproject.Paste()
+        temp_proj = ipkapp_temp.project_file
+        ipkapp_temp.close_project()
+        return temp_proj
+
+    @staticmethod
+    def __get_all_mapping(temp_proj):
```

Comment:
This method purpose seems a bit strange to me as it seems to return a mapping referencing itself `g[key_0][key_1] = g`.
Is this code valid ? If yes, what is its purpose ?

Reply:
The method reads the aedt file and extract the relevant info. Also transform the extracted list in an easy-to-use dictionary `{obj_name: props}`.

Reply:
Fine, thanks for the answer. It clarified the code content :)